testing/testrunner/utils.go

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: Elastic-2.0
+
+/*
+ * Copyright 2022 Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under
+ * one or more contributor license agreements. Licensed under the Elastic
+ * License 2.0; you may not use this file except in compliance with the Elastic
+ * License 2.0.
+ */
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"reflect"
+	"runtime"
+	"runtime/debug"
+	"syscall"
+)
+
+// This is a JSON type printed by the test binaries (not by EventsTrace), it's
+// used all over the place, so define it here to save space
+type TestPidInfo struct {
+	Tid  int64 `json:"tid"`
+	Tgid int64 `json:"tgid"`
+	Ppid int64 `json:"ppid"`
+	Pgid int64 `json:"pgid"`
+	Sid  int64 `json:"sid"`
+}
+
+// Definitions of types printed by EventsTrace for conversion from JSON
+type PidInfo struct {
+	Tid         int64 `json:"tid"`
+	Tgid        int64 `json:"tgid"`
+	Ppid        int64 `json:"ppid"`
+	Pgid        int64 `json:"pgid"`
+	Sid         int64 `json:"sid"`
+	StartTimeNs int64 `json:"start_time_ns"`
+}
+
+type CredInfo struct {
+	Ruid int64 `json:"ruid"`
+	Rgid int64 `json:"rgid"`
+	Euid int64 `json:"euid"`
+	Egid int64 `json:"egid"`
+	Suid int64 `json:"suid"`
+	Sgid int64 `json:"sgid"`
+}
+
+type TtyInfo struct {
+	Major int64 `json:"major"`
+	Minor int64 `json:"minor"`
+}
+
+type ProcessForkEvent struct {
+	ParentPids PidInfo `json:"parent_pids"`
+	ChildPids  PidInfo `json:"child_pids"`
+}
+
+type ProcessExecEvent struct {
+	Pids     PidInfo  `json:"pids"`
+	Creds    CredInfo `json:"creds"`
+	Ctty     TtyInfo  `json:"creds"`
```

Comment:
Struct field Ctty repeats json tag "creds":
```suggestion
	Creds    CredInfo `json:"creds"`
	Ctty     TtyInfo  `json:"ctty"`
```

Reply:
🙏